assets/js/frontend/quiz.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Functionality for quiz pages.
+ */
+
+jQuery( function() {
+
+	// Keep track of which input was used to submit the form. Note that if the user
+	// presses <Enter>, this click event should still be triggered.
+	jQuery( 'form[data-action-on-empty-response=warn] input[type=submit]' ).on( 'click', function( event ) {
+		jQuery( event.target ).attr( 'clicked', 'true' );
+	} );
+
+	// Warn on submitting empty responses. Note: using event delegation here to
+	// ensure that this callback happens *after* the WP Editor callbacks, so we
+	// have its value in the form data.
+	jQuery( 'html' ).on( 'submit', 'form[data-action-on-empty-response=warn]', function( event ) {
+		var form = jQuery( event.target );
+
+		// Get, and reset, the button that was clicked
+		var button = form.find( '[clicked=true]' );
+		button.removeAttr( 'clicked' );
+
+		// Only continue for the "Complete Quiz" button
+		if ( button.attr( 'name' ) != 'quiz_complete' ) {
```

Comment:
So if I understand this correctly - this should **not** show the modal when pressing the *Complete Quiz* button ?  
I'm asking as that still shows the modal for me and I'm actually wondering if we'd want such behaviour at all instead of always showing it when there are some empty fields ?

Reply:
Actually, this should *only* show the modal when the user clicks the _Complete Quiz_ button. If the button is not _Complete Quiz_, then we return here.

This is because the warning only makes sense (in my opinion) when the user is submitting the quiz to be graded. If they are clicking on _Reset Quiz_ or _Save Quiz_, for example, then I don't think we need to warn them of empty responses.

Does that make sense?